controllers/compassmanager_controller.go

```diff
@@ -284,6 +385,16 @@ func (cm *CompassManagerReconciler) needsToBeReconciled(obj runtime.Object) bool
 	return false
 }
 
+func (cm *CompassManagerReconciler) needsToBeDeleted(obj runtime.Object) bool {
```

Comment:
it may have been called that, but if you look at where the function is called, you'll notice that these are filter functions that trigger on specific events. The name needToBeDeleted clearly indicates which event we are dealing with in this function